ding/model/common/head.py

```diff
@@ -536,6 +536,113 @@ def forward(self, x: torch.Tensor) -> Dict:
+        )
+        self.V = nn.Sequential(
+            MLP(
+                hidden_size,
+                hidden_size,
+                hidden_size,
+                v_layer_num,
+                layer_fn=layer,
+                activation=activation,
+                norm_type=norm_type
+            ), block(hidden_size, 1)
+        )
+
+    def forward(self, s: torch.Tensor, a: torch.Tensor, mu_t: torch.Tensor, sigma_t: torch.Tensor,
+                sample_size: int = 10) -> Dict:
+        r"""
+        Overview:
+            Use encoded embedding tensor to predict Dueling output.
+            Parameter updates with DuelingHead's MLPs forward setup.
+        Arguments:
+            - s (:obj:`torch.Tensor`):
+                The encoded embedding state tensor, determined with given ``hidden_size``, i.e. ``(B, N=hidden_size)``.
+            - a (:obj:`torch.Tensor`):
+                The encoded embedding action tensor, determined with ``action_size``, i.e. ``(B, N=action_size)``.
+            - mu_t (:obj:`torch.Tensor`):
+                The mu output of actor head at timestep t, size (B, action_size)
+            - sigma_t (:obj:`torch.Tensor`):
+                The sigma output of actor head at timestep t, size (B, action_size)
+            - sample_size (:obj:`int`):
+                The num to samples for action when computing the Q value
+        Returns:
+            - outputs (:obj:`Dict`):
+                Run ``MLP`` with ``DuelingHead`` setups and return the result prediction dictionary.
+                Necessary Keys:
+                    - pred (:obj:`torch.Tensor`): Pred tensor of size ``(B, 1)``.
+        """
+        batch_size = s.shape[0]  # batch size * T
+        hidden_size = s.shape[1]
+        action_size = a.shape[1]  # TODO(pu): action dim is too small than hidden size
+        state_cat_action = torch.cat((s, a), dim=1)  # size (B, action_size + state_size)
+        a_val = self.A(state_cat_action)  # size (B, 1)
+        s_val = self.V(s)  # size (B,1)
+
+        mu_t = (torch.unsqueeze(mu_t, 1)).expand(
+            (batch_size, sample_size, action_size))  # size (B, sample_size, action_size)
+        sigma_t = (torch.unsqueeze(sigma_t, 1)).expand(
+            (batch_size, sample_size, action_size))  # size (B, sample_size, action_size)
+
+        expand_s = (torch.unsqueeze(s, 1)).expand(
+            (batch_size, sample_size, hidden_size))  # size (B, sample_size, hidden_size)
+        action_sample = torch.normal(mu_t, sigma_t)  # size (B, sample_size, action_size)
+
+        state_cat_action_sample = torch.cat((expand_s, action_sample),
+                                            dim=-1)  # size (B, sample_size, action_size + hidden_size)
+        a_val_sample = self.A(state_cat_action_sample)  # size (B, sample_size, 1)
+        a_val_sample = torch.squeeze(a_val_sample, -1)  # (B, sample_size)
+        logit = a_val - a_val_sample.mean(dim=-1, keepdim=True) + s_val  # size (B,1)
```

Comment:
rename `logit` to `q_value` can be better, do the same modification in `DuelingHead`